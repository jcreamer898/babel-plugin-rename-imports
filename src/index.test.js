import plugin from './index.js';
import pluginTester from 'babel-plugin-tester';

pluginTester({
    plugin,
    pluginOptions: {
        source: '@foo/path-utils',
        findName: 'join',
        replaceName: 'joinPath',
    },
    snapshot: true,
    tests: [
        `import {join} from '@foo/path-utils';
join('a','b');
notJoin();
`,
        `import {join as joinPath} from '@foo/path-utils';
joinPath('a','b');
`,
        `import * as pathUtils from '@foo/path-utils';
pathUtils.join('a','b');
pathUtils.notJoin();
`,
        `import {join as pathJoiner} from '@foo/path-utils';
pathJoiner('a','b');
`,
        `import {join} from 'not-path-utils';
join('a','b');
`,
        {
            title: 'codemod-should be configurable',
            pluginOptions: {
                source: 'baz',
                findName: 'foo',
                replaceName: 'bar',
            },
            code: `import {foo} from 'baz';
    foo('thing', true, 1234);
    `,
        },
        {
            title: 'Throws error with missing options',
            pluginOptions: {
                source: null,
                findName: null,
                replaceName: null,
            },
            snapshot: false,
            code: `import foo from 'bar';`,
            error: 'Error: babel-plugin-rename-imports - missing options',
        },
    ],
});
