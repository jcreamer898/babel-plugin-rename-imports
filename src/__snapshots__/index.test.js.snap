// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rename-imports 1. rename-imports: 1. rename-imports 1`] = `

import {join} from '@foo/path-utils';
join('a','b');
notJoin();


      ↓ ↓ ↓ ↓ ↓ ↓

import { joinPath } from "@foo/path-utils";
joinPath("a", "b");
notJoin();

`;

exports[`rename-imports 2. rename-imports: 2. rename-imports 1`] = `

import {join as joinPath} from '@foo/path-utils';
joinPath('a','b');


      ↓ ↓ ↓ ↓ ↓ ↓

import { joinPath } from "@foo/path-utils";
joinPath("a", "b");

`;

exports[`rename-imports 3. rename-imports: 3. rename-imports 1`] = `

import * as pathUtils from '@foo/path-utils';
pathUtils.join('a','b');
pathUtils.notJoin();


      ↓ ↓ ↓ ↓ ↓ ↓

import * as pathUtils from "@foo/path-utils";
pathUtils.joinPath("a", "b");
pathUtils.notJoin();

`;

exports[`rename-imports 4. rename-imports: 4. rename-imports 1`] = `

import {join as pathJoiner} from '@foo/path-utils';
pathJoiner('a','b');


      ↓ ↓ ↓ ↓ ↓ ↓

import { joinPath } from "@foo/path-utils";
joinPath("a", "b");

`;

exports[`rename-imports 5. rename-imports: 5. rename-imports 1`] = `

import {join} from 'not-path-utils';
join('a','b');


      ↓ ↓ ↓ ↓ ↓ ↓

import { join } from "not-path-utils";
join("a", "b");

`;

exports[`rename-imports 6. codemod-should be configurable: 6. codemod-should be configurable 1`] = `

import {foo} from 'baz';
    foo('thing', true, 1234);
    

      ↓ ↓ ↓ ↓ ↓ ↓

import { bar } from "baz";
bar("thing", true, 1234);

`;
